t/builtins/operators.t

```diff
@@ -771,6 +771,29 @@ use _007::Test;
         "divmodding by 0 is an error";
 }
 
+{
+    my $program = q:to/./;
+        say( 5 div 2 );
+        say( 5 div -2 );
+        .
+
+    outputs
+        $program,
+        "2\n-3\n",
+        "div operator (happy path)";
+}
+
+{
+    my $program = q:to/./;
+        say( 5 div 0 );
+        .
+
+    runtime-error
```

Comment:
Question: should it throw an error? Why not a warning and some equivalent of NaN? :) An error exists the program? Is it possible in 007, to catch an error och make the application only output an error but not the close the application?

Reply:
> Question: should it throw an error? Why not a warning and some equivalent of NaN?

Briefly, because `Int` throws an error in this case, and floating-point types (such as Java's `double`, or JavaScript's `number`) give a "special" numeric value back, such as `Inf` (in the case of `1 / 0`) or `NaN` (in the case of `0 / 0`).

While integer types _could_ be extended with infinities and a `NaN` value, this is basically never done. I suspect it's because [two's complement](https://en.wikipedia.org/wiki/Two%27s_complement) is such a representation sweet spot, so it wouldn't feel like an improvement. Also (if I understand things correctly), the fact that all major implementations of IEEE 457 choose not to throw an error/exception by default is a little bit of a historical accident, and not a super-happy one.

I expect we will have a `Real` type at some point (through a module), and I expect it will have `Inf`, `-Inf`, and `NaN`.

> An error exists the program? Is it possible in 007, to catch an error och make the application only output an error but not to close the application?

Yes, and ideally but not yet in this case. Two features need to be implemented for this to work: #65 (`CATCH` blocks), and #245. These in turn are blocking on some other things: #252 (in some form) and the [masak/new-object-system](https://github.com/masak/007/tree/masak/new-object-system) branch. So, uh, hang in there!